### ISSUE TITLE
Bash aliases set through module aren't escaped

### ISSUE THREAD

**Issue by 8573** [NixOS/nixpkgs#16973]
## Issue description

Bash aliases defined with the `programs.bash.shellAliases` configuration option are defined [(1)](https://github.com/NixOS/nixpkgs/blob/2974b6f4c8224ab1097e9386d6b3a0d570f95531/nixos/modules/programs/bash/bash.nix#L33) in `/etc/bashrc` as —

``` bash
alias ${k}='${v}'
```

— where `${k}` and `${v}` (which are variables in Nix, not in Bash) correspond to, respectively, the keys and values of the attribute set that is the value of `programs.bash.shellAliases`.

No escaping is performed on these values, so if one uses the apostrophe character in an alias defined in `programs.bash.shellAliases`, such as —

```
programs.bash.shellAliases = {
  "foo" = "foo --bar='baz qux'";
}
```

— then the `bash` module will emit the following problematic line in `/etc/bashrc` —

``` bash
alias foo='foo --bar='baz qux''
```

— which is equivalent to —

``` bash
alias foo='foo --bar=baz' qux
```

— which first defines an alias `foo`, then queries the alias `qux`. No such alias exists, so "bash: alias: qux: not found" is printed each time one starts an interactive Bash instance.
### Steps to reproduce

Use `programs.bash.shellAliases` to define an alias containing an apostrophe, as above.
## Technical details
- System: NixOS `16.09pre85931.125ffff (Flounder)`
- Nix version: `nix-env (Nix) 1.11.2`
- Nixpkgs version: `"16.09pre85931.125ffff"`

**Comment by 8573** [comment]
I confirm that I still can reproduce this issue, but (my apologies) I haven't time to try to fix it.

@nagisa: As you also encountered this issue, would you be willing to try to patch it?

**Comment by 8573** [comment]
I forgot to add the `nix-info -m` output for the system on which I reproduced the issue, which is as follows:
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.48, NixOS, 18.09pre143771.a8c71037e04 (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.0.4`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by 8573** [comment]
I think #47471 has fixed the same issue for zsh.

**Comment by timokau** [comment]
Fixed by @8573 in #47984.

**Comment by timokau** [comment]
Because you have to use certain keywords github understands in either the PR description or the commit message, for example "Fixes #16973" would work.

**Comment by timokau** [comment]
True, that is weird.
